R/tm_a_pca.R

```diff
@@ -18,45 +18,84 @@
 #' @templateVar ggnames "Elbow plot", "Circle plot", "Biplot", "Eigenvector plot"
 #' @template ggplot2_args_multi
 #'
-#' @export
-#'
 #' @examples
+#' # general data example
+#' library(teal.widgets)
 #'
-#' # ADSL example
+#' data <- teal_data()
+#' data <- within(data, {
+#'   library(nestcolor)
+#'   USArrests <- USArrests
+#' })
+#' datanames(data) <- "USArrests"
+#'
+#' app <- init(
+#'   data = data,
+#'   modules = modules(
+#'     tm_a_pca(
+#'       "PCA",
+#'       dat = data_extract_spec(
+#'         dataname = "USArrests",
+#'         select = select_spec(
+#'           choices = variable_choices(
+#'             data = data[["USArrests"]], c("Murder", "Assault", "UrbanPop", "Rape")
+#'           ),
+#'           selected = c("Murder", "Assault"),
+#'           multiple = TRUE
+#'         ),
+#'         filter = NULL
+#'       ),
+#'       ggplot2_args = ggplot2_args(
+#'         labs = list(subtitle = "Plot generated by PCA Module")
+#'       )
+#'     )
+#'   )
+#' )
+#'
+#' if (interactive()) {
+#'   shinyApp(app$ui, app$server)
+#' }
+#'
+#' # CDISC data example
+#' library(teal.widgets)
 #'
 #' data <- teal_data()
 #' data <- within(data, {
 #'   library(nestcolor)
-#'   ADSL <- teal.modules.general::rADSL
```

Comment:
hi @chlebowa I know that we try to strive from prefixes in examples. The main reason I put prefixes before data name in examples was that `rADSL` exists in `teal.transform` and some other teal. packages (can't recall which others now, but `teal.transform` for sure). And the prefix is in here so that to make sure you use `rADSL` from `teal.modules.general` and not from other package, if you attached other package after you attached `teal.modules.general`.

Reply:
1. Are `rADSL` objects in other packages different or are they the same object exported from multiple packages?
2. When running an example, `teal.modules.general` will presumably attached last.Specifically, it is impossible to have `teal.transform` attached before `teal.modules.general` because the latter `Depends` on the former.

Reply:
You can resolve, once you are done reading my lengthy reply :P